pkg/webhook/cm/plugins/sloconfig/resource_qos_checker.go

```diff
@@ -111,6 +115,254 @@ func CheckNetQosCfg(cfg *configuration.ResourceQOSCfg, fldPath *field.Path) erro
+			}
+		}
+		if nodeStrategy.BEClass != nil && nodeStrategy.BEClass.NetworkQOS != nil {
+			if nodeStrategy.BEClass.NetworkQOS.IngressRequest != nil && nodeStrategy.BEClass.NetworkQOS.IngressRequest.Type == intstr.String {
+				ingressRequestTotal.Sub(resource.MustParse(nodeStrategy.BEClass.NetworkQOS.IngressRequest.StrVal))
+			}
+			if nodeStrategy.BEClass.NetworkQOS.EgressRequest != nil && nodeStrategy.BEClass.NetworkQOS.EgressRequest.Type == intstr.String {
+				egressRequestTotal.Sub(resource.MustParse(cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest.StrVal))
+			}
+		}
+		if ingressRequestTotal.Value() < 0 || egressRequestTotal.Value() < 0 {
+			return fmt.Errorf("total net bandwidth quantity already over total network bandwidth in cfg.NodeStrategies[%d]", idx)
+		}
+	}
+
+	return nil
+}
+
+func getTotalNetBandWidth(labelSelector *metav1.LabelSelector, sysCfg *configuration.SystemCfg) *resource.Quantity {
+	res := resource.NewQuantity(0, resource.DecimalSI)
+	if sysCfg == nil {
+		return res
+	}
+
+	if len(sysCfg.NodeStrategies) == 0 && sysCfg.ClusterStrategy != nil {
+		return &sysCfg.ClusterStrategy.TotalNetworkBandwidth
+	}
+
+	for _, cur := range sysCfg.NodeStrategies {
+		if reflect.DeepEqual(labelSelector, cur.NodeSelector) {
+			return &cur.TotalNetworkBandwidth
+		}
+	}
+
+	return res
+}
+
+func CheckOnlyOneFormat(cfg *configuration.ResourceQOSCfg) error {
+	if cfg == nil {
+		return nil
+	}
+
+	// check netqos config is valid for cluster strategy
+	if cfg.ClusterStrategy != nil {
+		allTypes := make(map[intstr.Type]interface{})
+		if cfg.ClusterStrategy.LSRClass != nil && cfg.ClusterStrategy.LSRClass.NetworkQOS != nil {
+			if cfg.ClusterStrategy.LSRClass.NetworkQOS.IngressRequest != nil {
+				allTypes[cfg.ClusterStrategy.LSRClass.NetworkQOS.IngressRequest.Type] = nil
+			}
+			if cfg.ClusterStrategy.LSRClass.NetworkQOS.EgressRequest != nil {
+				allTypes[cfg.ClusterStrategy.LSRClass.NetworkQOS.EgressRequest.Type] = nil
+			}
+		}
+		if cfg.ClusterStrategy.LSClass != nil && cfg.ClusterStrategy.LSClass.NetworkQOS != nil {
+			if cfg.ClusterStrategy.LSClass.NetworkQOS.IngressRequest != nil {
+				allTypes[cfg.ClusterStrategy.LSClass.NetworkQOS.IngressRequest.Type] = nil
+			}
+			if cfg.ClusterStrategy.LSClass.NetworkQOS.EgressRequest != nil {
+				allTypes[cfg.ClusterStrategy.LSClass.NetworkQOS.EgressRequest.Type] = nil
+			}
+		}
+		if cfg.ClusterStrategy.BEClass != nil && cfg.ClusterStrategy.BEClass.NetworkQOS != nil {
+			if cfg.ClusterStrategy.BEClass.NetworkQOS.IngressRequest != nil {
+				allTypes[cfg.ClusterStrategy.BEClass.NetworkQOS.IngressRequest.Type] = nil
+			}
+			if cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest != nil {
+				allTypes[cfg.ClusterStrategy.BEClass.NetworkQOS.EgressRequest.Type] = nil
+			}
+		}
+
+		if len(allTypes) > 1 {
+			return errors.New("only one type can be used for percentages or quantity in clusterStrategy")
```

Comment:
Here, errors is the package `github.com/pkg/errors`, maybe a typo?  Should use `"errors"`.

Reply:
done